frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +537,138 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
+  wr.heading(kind_, type_, loc,
+             "type in 'new' expression is missing its argument list.");
+}
+
+void ErrorUseImportNeedsModule::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto isImport = std::get<bool>(info);
+  std::string useOrImport = isImport ? "import" : "use";
+  wr.heading(kind_, type_, loc, "'", useOrImport,
+             "' statements must refer to module", (isImport ? "" : " or enum"),
+             " symbols.");
+  wr.message("(e.g., '", useOrImport, " <module>[.<submodule>]*;')");
+}
+
+void ErrorExceptOnlyInvalidExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto limitationKind = std::get<uast::VisibilityClause::LimitationKind>(info);
+  wr.heading(kind_, type_, loc, "incorrect expression in '", limitationKind,
+             "' list, identifier expected.");
+}
+
+void ErrorLabelIneligibleStmt::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto maybeStmt = std::get<const uast::AstNode*>(info);
+  std::string msg = "cannot label";
+  if (maybeStmt) msg = msg + " '" + tagToString(maybeStmt->tag()) + "'";
+  msg += " statement.";
+  wr.heading(kind_, type_, loc, msg);
+  wr.message("Only for-, while-do- and do-while-statements can have labels.");
+}
+
+void ErrorBisonMemoryExhausted::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "memory exhausted while parsing.");
+}
+
+void ErrorBisonUnknownError::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<1>(info);
+  auto nearestToken = std::get<2>(info);
+  wr.heading(kind_, type_, loc,
+             "unknown error from Bison parser: ", errorMessage, ".");
+  if (!nearestToken.empty()) {
+    wr.note(loc, "Nearest token when error was encountered: '", nearestToken,
+            "'");
+  }
+}
+
+void ErrorBisonSyntaxError::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto nearestToken = std::get<std::string>(info);
+  wr.heading(kind_, type_, loc, "syntax error",
+             (nearestToken.empty() ? "" : " near '" + nearestToken + "'"));
+}
+
+/* lexer errors */
+
+void ErrorStringLiteralEOF::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc, "end-of-file in string literal.");
+}
+
+void ErrorExternUnclosedPair::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto pairSymbol = std::get<std::string>(info);
+  wr.heading(kind_, type_, loc, "missing closing ", pairSymbol,
+             " symbol in extern block.");
+}
+
+void ErrorCommentEOF::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<0>(info);
+  auto startLoc = std::get<1>(info);
+  auto nestedLoc = std::get<2>(info);
+  wr.heading(kind_, type_, loc, "EOF in comment.");
+  wr.note(startLoc, "Unterminated comment started here:");
```

Comment:
Aren't notes uncapitalized by default, and uppercased automatically? should this be `unterminated`?